pkg/operator/options/options_validation.go

```diff
@@ -31,11 +31,21 @@ func (o Options) Validate() error {
 		o.validateRequiredFields(),
 		o.validateEndpoint(),
 		o.validateVMMemoryOverheadPercent(),
+		o.validateNetworkPluginMode(),
 		o.validateVnetSubnetID(),
 		validate.Struct(o),
 	)
 }
 
+func (o Options) validateNetworkPluginMode() error {
+	// TODO: Move overlay and none to shared constants in AgentBaker
+	// NOTE: Network Plugin Mode should be normalized on the AKS API Level, so if we are passing values in from the ManagedCluster, NetworkPluginMode will already be normalized.
+	if o.NetworkPluginMode != "overlay" && o.NetworkPluginMode != "" && o.NetworkPluginMode != "none" {
```

Comment:
we can remove "none" I validated we no longer support this as a possible input into the NetworkPluginMode AKS API.